### PR TITLE
Build and push multus container to quay.io

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
We are hiting rate limit from docker.io, this change build and push the multus
container to quay.io.

**Special notes for your reviewer**:

**Release note**:

```release-note
Move multus container to quay.io
```

### CODE REVIEW

File: hack/components/bump-multus.sh [hunk 1]
```diff
@@ -106,10 +106,13 @@ rm -rf data/multus/*
 cp ${MULTUS_PATH}/config/cnao/000-ns.yaml data/multus/
 cp ${MULTUS_PATH}/config/cnao/001-multus.yaml data/multus/
 
-echo 'Get multus image name and update it under CNAO'
+echo 'Get multus image, push it to quay.io and update it under CNAO'
 MULTUS_TAG=$(git-utils::get_component_tag ${MULTUS_PATH})
-MULTUS_IMAGE=nfvpe/multus
+MULTUS_IMAGE=kubevirt/multus
```
Comment (phoracek) [inline hack/components/bump-multus.sh:111]: Use our existing repo instead https://quay.io/repository/kubevirt/cluster-network-addon-multus

File: hack/components/bump-multus.sh [hunk 2]
```diff
@@ -106,11 +106,47 @@ rm -rf data/multus/*
...
+
+# This dockerfile is specific to building Multus for OpenShift
+FROM openshift/origin-release:golang-1.15 as builder
+
+ADD . /usr/src/multus-cni
+
+WORKDIR /usr/src/multus-cni
+RUN ./build
+
+# This Dockerfile is used to build the image available on DockerHub
```
Comment (phoracek) [inline hack/components/bump-multus.sh:125]: Do we need this invalid comment?

File: hack/components/bump-multus.sh [hunk 3]
```diff
@@ -106,11 +106,47 @@ rm -rf data/multus/*
...
 MULTUS_TAG=$(git-utils::get_component_tag ${MULTUS_PATH})
-MULTUS_IMAGE=nfvpe/multus
+MULTUS_IMAGE=quay.io/kubevirt/cluster-network-addon-multus
 MULTUS_IMAGE_TAGGED=${MULTUS_IMAGE}:${MULTUS_TAG}
-MULTUS_IMAGE_DIGEST="$(docker-utils::get_image_digest "${MULTUS_IMAGE_TAGGED}" "${MULTUS_IMAGE}")"
+(
+    cd ${MULTUS_PATH}
+    cat <<EOF > Dockerfile
+
+# This dockerfile is specific to building Multus for OpenShift
```
Comment (phoracek) [inline hack/components/bump-multus.sh:117]: But this image will be never used on OpenShift. Could we just drop the misleading comment?

**Review by phoracek — CHANGES_REQUESTED** [review]
How do we test this? e2e tests on CI are running with the original Multus image, don't they? We have to somehow verify that this new container delivers correct binaries on the right place.

**Comment by qinqon** [comment]
/override pull-e2e-cluster-network-addons-operator-nmstate-functests

not related

**Comment by qinqon** [comment]
> How do we test this? e2e tests on CI are running with the original Multus image, don't they? We have to somehow verify that this new container delivers correct binaries on the right place.

What do you mean ? the CI tests use the image from CNAO wich is the one I have manually pushed with the output of "bump-multus" so we are really testing it.

File: pkg/components/components.go [hunk 4]
```diff
@@ -402,12 +402,12 @@ func GetCrd() *extv1.CustomResourceDefinition {
 		Properties: map[string]extv1.JSONSchemaProps{
 			"matchExpressions": extv1.JSONSchemaProps{
 				Description: "matchExpressions is a list of label selector requirements. The requirements are ANDed.",
-				Type: "array",
+				Type:        "array",
```
Comment (phoracek) [inline pkg/components/components.go:405]: Don't we have gofmt running on our CI? How did these issues got here...

**Review by phoracek — APPROVED** [review]
Ack

/lgtm
/approve

**Comment by qinqon** [comment]
/override pull-e2e-cluster-network-addons-operator-nmstate-functests

**Pull request merged**